Add release notes for OEE 10.17 (off cycle). (#281)
SoftwareAG/c8y-releasenotes: content/release-10-17-0/oee-10-17-0.md

---
weight: 95
title: OEE App
layout: bundle
---

### Improvements

<table ><colgroup>
<col style="width: 17%;">
<col style="width:57%;">
<col style="width: 11%;">
<col style="width: 15%;">
</colgroup>
<thead><tr>
<th>
Product area</th>
<th>
Description</th>
<th>
Issue</th>
<th>Build comp.</th>
</tr>
</thead><tbody>

<tr>
<td>Administration</td>

> <td>This release of the Cumulocity IoT OEE Application adds the possibility to calculate the OEE value based on defined shift plans. A new and improved user interface is available to allow OEE Administrators to add, edit and remove shift plans.</td>

Reviewer: Can you check on the capitalization of "Cumulocity IoT Application" and make it consistent across the release notes? Sometimes the "application" part is spelt with lower case "a".

Reviewer 2: Should be spelled with lower case consistently. 

Also change Administrator > administrator

Author: Done.

<td>COE-549</td>
<td>UI</td>

> </tr>

Reviewer: I would suggest adding the following information:
* technically it was already possible to define shift plans before but 
* we did not have a UI
* you have to create individual shifts, now it is possible to define recurring shifts
* In additional to OEE calculations for intervals, the OEE App now calculates the OEE also for whole shifts

Author: Done.

<tr>
<td>Product experience</td>

> <td>Updated the WebSDK to 10.17 in order to adopt the new Cumulocity UI.</td>

Reviewer 2: ```suggestion
<td>Updated the Web SDK to 10.17 in order to adopt the new Cumulocity IoT UI.</td>
```

<td>COE-445</td>
<td>UI</td>
</tr>

</tbody></table>

### Fixes

<table >
<colgroup>
<col style="width: 17%;">
<col style="width:57%;">
<col style="width: 11%;">
<col style="width: 15%;">
</colgroup>
<thead><tr>
<th>
Product area</th>
<th>
Description</th>
<th>
Issue</th>
<th>Build comp.</th>
</tr>
</thead><tbody>

<tr>
<td>Product experience</td>

> <td>Previously, locations given in the Filter Location dropdown in the Profiles page were not sorted. They are now sorted alpabetically in order to improve findability. </td>

Reviewer 2: ```suggestion
<td>Previously, locations given in the <b>Filter location</b> dropdown in the <b>Profiles</b> page were not sorted. They are now sorted alphabetically in order to improve findability. </td>
```

<td>COE-884</td>
<td>UI</td>
</tr>

<tr>
<td>Administration</td>

> <td>Improve the behaviour of the Microservices during startup. Includes fixes for an issue that caused "Permission Denied" messages in certain configurations.</td>

Reviewer 2: ```suggestion
<td>Improved the behaviour of the microservices during startup. Includes fixes for an issue that caused "Permission Denied" messages in certain configurations.</td>
```

<td>COE-857</td>
<td>Microservice</td>
</tr>

<tr>
<td>Administration</td>
<td>Improvements to the handling of the log files created by the microservice.</td>
<td>COE-777</td>
<td>Microservice</td>
</tr>

<tr>
<td>Product experience</td>

> <td>Removed obsolete Section View from the Machine Dashboard.</td>

Reviewer 2: ```suggestion
<td>Removed obsolete <b>Section</b> view from the <b>Machine dashboard</b>.</td>
```

<td>COE-735</td>
<td>UI</td>
</tr>

<tr>
<td>Product experience</td>
<td>Improved the rendering behaviour of webpages when the user's language is set to Japanese.</td>
<td>COE-732</td>
<td>UI</td>
</tr>

<tr>
<td>Product experience</td>

> <td>Add more translated strings to improve localisation.</td>

Reviewer 2: ```suggestion
<td>Added more translated strings to improve the localization.</td>
```

<td>COE-681</td>
<td>UI</td>
</tr>

<tr>
<td>Product experience</td>

> <td>The behaviour of the Organisation Overview page was updated. It will now handle cases where OEE profiles are missing or removed in a more user-friendly and obvious way.</td>

Reviewer 2: ```suggestion
<td>The behaviour of the <b>Organization overview</b> page was updated. It now handles cases where OEE profiles are missing or removed in a more user-friendly and obvious way.</td>
```

Reviewer 2: Is that the name of the page? Isn´t it just called "Overview"?

Author: Yes. ln the UI, it's always called Overview.

Author: ```suggestion
<td>The behaviour of the <b>Overview</b> page was updated. It now handles cases where OEE profiles are missing or removed in a more user-friendly and obvious way.</td>
```

<td>COE-677</td>
<td>UI</td>
</tr>

<tr>
<td>REST API</td>
<td>Improved the behaviour of the microservice when REST calls are made with non-existant profile IDs. </td>
<td>COE-673, COE-663</td>
<td>Microservice</td>
</tr>

<tr>
<td>OEE calculation</td>

> <td>Fixed an issue that occurred with OEE calculation when the same event is used for APT and APA types.</td>

Reviewer: should we write "actual production time" and "actual production amount"

Author: I think so. Done.

<td>COE-671</td>
<td>Microservice</td>
</tr>

<tr>
<td>Administration</td>

> <td>Fixed some issues that could cause background processes to continue running when tenant is unsubscribed from the OEE microservices.</td>

Reviewer 2: ```suggestion
<td>Fixed some issues that could cause background processes to continue running when the tenant is unsubscribed from the OEE microservices.</td>
```

> <td>Fixed an issue that occurred with OEE calculation when the same event is used for <b>actual production time</b> (APT) and <b>actual production amount</b> (APA) types.</td>

Reviewer 3: ```suggestion
<td>Fixed an issue that occurred with OEE calculation when the same event is used for <b>Actual production time</b> (APT) and <b>Actual production amount</b> (APA) types.</td>
```

<td>COE-667, COE-662</td>
<td>Microservice</td>
</tr>

<tr>
<td>REST API</td>

> <td>Ensured that REST calls to <code>service/oee-bundle/components</code> returns the correct data, rather than an empty array.</td>

Reviewer 2: ```suggestion
<td>Ensured that REST calls to <code>service/oee-bundle/components</code> return the correct data, rather than an empty array.</td>
```

<td>COE-666</td>
<td>Microservice</td>
</tr>

<tr>
<td>Product experience</td>

> <td>Small graphical alignment fixes for the Calculation Categories page.</td>

Reviewer 2: ```suggestion
<td>Small graphical alignment fixes for the <b>Calculation categories</b> page.</td>
```
